cave/utils/csv2rh.py

```diff
@@ -0,0 +1,250 @@
+                    column_name.startswith('p_') or  # parameter
+                    column_name.startswith('i_')):  # instance feature
+                raise ValueError("%s not a legal column name in %s." % column_name, csv_path)
+
+        self._complete_configs()
+        self._complete_instances()
+        self.logger.debug("Found: seed=%s, cost=%s, time=%s, status=%s",
+                          'seed' in self.data.columns, 'cost' in self.data.columns,
+                          'time' in self.data.columns, 'status' in self.data.columns)
+
+        # Create RunHistory
+        rh = RunHistory(average_cost)
+        def add_to_rh(row):
+            new_status = self._interpret_status(row['status']) if 'status' in row else StatusType.SUCCESS
+            rh.add(config=self.id_to_config[row['config_id']],
+                   cost=row['cost'],
+                   time=row['time'] if 'time' in row else -1,
+                   status=new_status,
+                   instance_id=row['instance_id'] if 'instance_id' in row else None,
+                   seed=row['seed'] if 'seed' in row else None,
+                   additional_info=None,
+                   origin=DataOrigin.INTERNAL)
+
+        self.data.apply(add_to_rh, axis=1)
+        return rh
+
+    def get_cs(self):
+        if self.pcs and type(self.pcs) == ConfigurationSpace:
+            cs = self.pcs
+        elif self.pcs and type(self.pcs) == str:
+            parameters = [p[2:] for p in self.data.columns if p.startswith('p_')]
+            with open(self.pcs, 'r') as fh:
+                cs = pcs.read(fh)
+            diff = set(cs.get_hyperparameter_names()).symmetric_difference(set(parameters))
+            if not len(diff) == 0:
+                raise ValueError("Comparing parameters from \"%s\" and csv. "
+                                 "In pcs but not in csv: [%s]. "
+                                 "In csv but not in pcs: [%s]. " % (self.pcs,
+                                 str(set(cs.get_hyperparameter_names()).difference(set(parameters))),
+                                 str(set(parameters).difference(set(cs.get_hyperparameter_names())))))
+        else:
+            parameters = [p for p in self.data.columns if p.startswith('p_')]
+            self.logger.warning("No parameter configuration space (pcs) provided! "
+                                "Interpreting all parameters as floats. This might lead "
+                                "to suboptimal analysis.")
+            minima = self.data.min()  # to define ranges of hyperparameters
+            maxima = self.data.max()
+            cs = ConfigurationSpace(seed=42)
+            for p in parameters:
+                cs.add_hyperparameter(UniformFloatHyperparameter(p[2:],
+                                      lower=minima[p] - 1, upper=maxima[p] + 1))
+        self.cs = cs
+
+    def _interpret_status(self, status):
+        if status.upper() in ["SAT", "UNSAT", "SUCCESS"]:
```

Comment:
Are we sure that status will never contain white-space? If not we should check if status.strip().upper() is in [XYZ]

Reply:
Jup, done.